docs/proposals/resource-catalog.md

```diff
@@ -0,0 +1,222 @@
+
+
+We can reshape the catalog resource to more easily express them, secure them, and claim them.
+
+# Goals
+ 1.  cluster-admin should control which catalog exist.
+ 2.  cluster-admin should be able/disable per-catalog exposure on a per-project granularity.
+ 3.  project-admin should be able to delegate per-catalog exposure.
+ 4.  project-editor should be able to claim an entry from the catalog.
+ 5.  ResourceCatalogClaims should be sync-able, but not require syncing.
+ 6.  project-admin should be able to control the shape of claim realization and prevent unwanted injection.
```

Comment:
> Define unwanted injection

Project admin may want to claim something from catalog, but want to make sure that it can't create serviceaccounts, secrets, or some other resource type in his namespace.